server/app/uk/gov/ons/addressIndex/server/model/response/AddressResponse.scala

```diff
@@ -0,0 +1,251 @@
+  */
+case class AddressBySearchResponseContainer(
+  response: AddressResponse,
+  status: AddressResponseStatus,
+  errors: Seq[AddressResponseError]
+)
+
+/**
+  * Contains relevant, to the address request, data
+  *
+  * @param tokens    address decomposed into relevant parts (building number, city, street, etc.)
+  * @param addresses found addresses
+  * @param limit     max number of found addresses
+  * @param offset    offset of found addresses (for pagination)
+  * @param total     total number of found addresses
+  */
+case class AddressResponse(
+  tokens: AddressTokens,
+  addresses: Seq[AddressResponseAddress],
+  limit: Int,
+  offset: Int,
+  total: Int
+)
+
+/**
+  * Contains tokens that build that the address can be splitted onto
+  *
+  * @param uprn           uprn
+  * @param buildingNumber building number
+  * @param postcode       postcode
+  */
+case class AddressTokens(
+  uprn: String,
+  buildingNumber: String,
+  postcode: String
+)
+
+/**
+  * Contains address information retrieved in ES (PAF or NAG)
+  *
+  * @param uprn               uprn
+  * @param formattedAddress   cannonical address form
+  * @param paf                optional, information from Paf index
+  * @param nag                optional, information from Nag index
+  * @param geo                optional, geo information
+  * @param underlyingScore    score from elastic search
+  * @param underlyingMaxScore maximum score in the elastic result
+  */
+case class AddressResponseAddress(
+  uprn: String,
+
+  formattedAddress: String,
+
+  paf: Option[AddressResponsePaf],
+  nag: Option[AddressResponseNag],
+
+  // geo from NAG, null if address only in PAF (i.e. not found in NAG)
+  geo: Option[AddressResponseGeo],
+
+  underlyingScore: Float, // from Elastic
+  underlyingMaxScore: Float // from Elastic (repeated for each address)
+
+)
+
+object AddressResponseAddress {
+  /**
+    * Transforms Paf address from elastic search into the Response address
+    *
+    * @param other
+    * @return
+    */
+  def fromPafAddress(other: PostcodeAddressFileAddress): AddressResponseAddress =
+  AddressResponseAddress(
+    uprn = other.uprn,
+    formattedAddress = "",
+    paf = Some(AddressResponsePaf(
+      other.udprn,
+      other.organizationName,
+      other.departmentName,
+      other.subBuildingName,
+      other.buildingName,
+      other.buildingNumber,
+      other.dependentThoroughfare,
+      other.thoroughfare,
+      other.doubleDependentLocality,
+      other.dependentLocality,
+      other.postTown,
+      other.postcode,
+      other.postcodeType,
+      other.deliveryPointSuffix,
+      other.welshDependentThoroughfare,
+      other.welshThoroughfare,
+      other.welshDoubleDependentLocality,
+      other.welshDependentLocality,
+      other.welshPostTown,
+      other.poBoxNumber,
+      other.startDate,
+      other.endDate
+    )),
+    nag = None,
+    geo = None,
+    underlyingScore = 1,
+    underlyingMaxScore = 1
```

Comment:
These will be set to the raw Elastic scores for demonstration purposes. It's extremely likely that we'll replace these two fields with something else - heck, we might even rename them to _underlyingXxx and add two new fields score and maxScore. I wouldn't worry about it. 'Root' makes them sound more important than they actually are.